src/DotVVM.Framework/Compilation/Validation/DefaultControlUsageValidator.cs

```diff
@@ -59,12 +59,15 @@ public IEnumerable<ControlUsageError> Validate(IAbstractControl control)
 
         protected virtual MethodInfo[] FindMethods(Type type)
         {
+            if (type == typeof(object)) return new MethodInfo[0];
             var methods = type.GetMethods(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static)
                 .Where(m => m.IsDefined(typeof(ControlUsageValidatorAttribute)))
                 .ToArray();
-            if (methods.Length > 0) return methods;
-            else if (type == typeof(object)) return new MethodInfo[0];
-            else return FindMethods(type.GetTypeInfo().BaseType);
+
+            var overrideValidation = methods.Select(s => s.GetCustomAttribute(typeof(ControlUsageValidatorAttribute))).Any(s => ((ControlUsageValidatorAttribute)s).Override);
```

Comment:
It's probably ok that this is a breaking change, but we should make sure that all declarations are consistent in the `Override` property. It's not very intuitive, what this should do (if not error):

```
[ControlUsageValidatorAttribute(Override = true)
Method1()...

[ControlUsageValidatorAttribute(Override = false)
Method2()...
```

Reply:
I have added the consistency check.